### PR TITLE
feat: [Fc-2] Show in comming message

### PR DESCRIPTION
**Pull request by huydosgtech**

### CODE REVIEW

**Comment by huydosgtech** [comment]
Video demo: 

https://github.com/saigontechnology/rn-firebase-chat/assets/105196859/ccb2ffa5-3c39-4c85-9e63-d82ff1782c5d


https://github.com/saigontechnology/rn-firebase-chat/assets/105196859/a9a05a7e-81a2-4d75-8e92-4c505ac53533

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: src/reducer/action.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import type { ConversationProps } from '../interfaces';
 export enum ChatActionKind {
   SET_LIST_CONVERSATION = 'SET_LIST_CONVERSATION',
   SET_CONVERSATION = 'SET_CONVERSATION',
+  UPDATE_USER_CONVERSATION = 'UPDATE_USER_CONVERSATION',
```
[the remark this answers is not in the document]
Comment (huydosgtech) [reply src/reducer/action.ts]: yes, I updated it. Please check again

File: src/chat/ChatScreen.tsx [hunk 2]
```diff
@@ -96,6 +97,31 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
...
+          const userInfoIncomming = {
+            id: message.id,
+            name: message.senderId,
+          } as IUserInfo;
+          const formatMessage = formatMessageData(message, userInfoIncomming);
+          const mergeMessageList = [
+            formatMessage,
+            ...messageRef.current,
+          ] as MessageProps[];
+          setMessagesList(mergeMessageList);
```
[the remark this answers is not in the document]
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: yes, I updated it. Please check again

File: src/reducer/action.ts [hunk 3]
```diff
@@ -14,3 +15,8 @@ export const setConversation = (payload: ConversationProps) => ({
   type: ChatActionKind.SET_CONVERSATION,
   payload,
 });
+
+export const updateUserConversation = (payload: ConversationProps) => ({
```
[the remark this answers is not in the document]
Comment (huydosgtech) [reply src/reducer/action.ts]: yes, I updated it. Please check again

File: src/reducer/chat.ts [hunk 4]
```diff
@@ -26,5 +26,17 @@ export const chatReducer = (
         ...state,
         conversation: action.payload as ConversationProps,
       };
+    case ChatActionKind.UPDATE_USER_CONVERSATION:
+      const message = action.payload as ConversationProps;
+      const newListMessage = (state.listConversation || [])
```
[the remark this answers is not in the document]
Comment (huydosgtech) [reply src/reducer/chat.ts]: I updated, please help me check it. Thank you

https://github.com/saigontechnology/rn-firebase-chat/assets/105196859/0aa65834-6a54-4058-a998-3f2db82af622

File: src/chat/ChatScreen.tsx [hunk 5]
```diff
@@ -96,6 +99,39 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
     [firebaseInstance, memberIds, partners]
   );
 
+  useEffect(() => {
+    return () => {
+      firebaseInstance.clearConversationInfo();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (honghoangsts) [inline src/chat/ChatScreen.tsx:106]: remove eslint-disable-next-line react-hooks/exhaustive-deps

Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: I updated

**Pull request merged**